src/gc/unix/gcenv.unix.cpp

```diff
@@ -722,6 +751,58 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
     return cacheSize;
 }
 
+// Get memory size multiplier based on the passed in units (k = kilo, m = mega, g = giga)
+static uint64_t GetMemorySizeMultiplier(char units)
+{
+    switch(units)
+    {
+        case 'g':
+        case 'G': return 1024 * 1024 * 1024;
+        case 'm':
+        case 'M': return 1024 * 1024;
+        case 'k':
+        case 'K': return 1024;
+    }
+
+    // No units multiplier
+    return 1;
+}
+
+#ifndef __APPLE__
+// Try to read the MemAvailable entry from /proc/meminfo.
+// Return true if the /proc/meminfo existed, the entry was present and we were able to parse it.
+static bool ReadMemAvailable(uint64_t* memAvailable)
+{
+    bool foundMemAvailable = false;
+    FILE* memInfoFile = fopen("/proc/meminfo", "r");
```

Comment:
While this is fine -- you could also use `sysinfo(2)` here, which obtains the information from the same place that `procfs` uses to populate `/proc/meminfo`, but should work where `procfs` isn't available for whatever reason.

Reply:
Oh, nevermind, saw it's being used down the line. :)

Reply:
Actually, we use sysinfo only to extract the swap size. But looking at the sysinfo, it doesn't seem to include the equivalent of the MemAvailable.